jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/BlobPullerIntegrationTest.java

```diff
@@ -74,10 +71,13 @@ public void testPull_unknownBlob()
           RegistryClient.factory(EVENT_EMITTER, "localhost:5000", "busybox")
               .setAllowInsecureRegistries(true)
               .newRegistryClient();
-      registryClient.pullBlob(nonexistentDigest, Mockito.mock(OutputStream.class));
+      registryClient.pullBlob(nonexistentDigest).writeTo(ByteStreams.nullOutputStream());
       Assert.fail("Trying to pull nonexistent blob should have errored");
 
-    } catch (RegistryErrorException ex) {
+    } catch (IOException ex) {
+      if (!(ex.getCause() instanceof RegistryErrorException)) {
+        throw ex;
```

Comment:
This is here because the call should only be throwing an `IOException` whose cause is a `RegistryErrorException`. Otherwise, we just throw the original exception. I chose this in favor of asserting the cause type since this gives the full exception if something did go wrong.

Reply:
Ah, I missed its doing `getClause()`.

Unrelated, but this test should actually narrow the scope of the `try` block to wrap only the method that actually throws the expected exception. That's the main reason we don't use `@Test(expected=...)`.